monkeytypes/errors.py

```diff
@@ -0,0 +1,6 @@
+class IllegalMutationError(RuntimeError):
+    """
+    Raised when an error occurs during illegal mutation of fields
+    """
+
+    pass
```

Comment:
Let's stash this in base_models.py. I don't think it needs its own file, and moving it to base_models add some context around what it's for. External users won't be affected since it's exported in __init__.py.